src/qibolab/instruments/qm/program/acquisition.py

```diff
@@ -9,8 +9,25 @@
 from qualang_tools.addons.variables import assign_variables_to_element
 from qualang_tools.units import unit
 
-from qibolab.execution_parameters import AcquisitionType, AveragingMode
-from qibolab.qubits import QubitId
+from qibolab.execution_parameters import (
+    AcquisitionType,
+    AveragingMode,
+    ExecutionParameters,
+)
+from qibolab.result import collect
+
+
+def _split(data, npulses, iq=False):
+    """Split results of different readout pulses to list.
+
+    These results were acquired in the same acquisition stream in the
+    instrument.
+    """
+    if npulses == 1:
+        return [data]
+    if iq:
+        return [data[..., i, :] for i in range(npulses)]
```

Comment:
```suggestion
        return data[..., :npulses, :]
```
?

This isn't the exact same, since it's transposed, and the result is an array, and not a list.

The exact equivalent is:
```suggestion
        return list(np.moveaxis(data, -2, 0)[:npulses])
```
but, if the dimension contains exactly `npulses` elements, you could even skip the further `[:npulses]`.

Reply:
Initially, I wrote a more complicate expression with `np.transpose()`, but I was sure there should have been an easier way.

And now that I found `np.moveaxis()`, I should also use it for:
https://github.com/qiboteam/qibolab/blob/1e3a501aa43f907a2415f1ebe903c3853a093a38/src/qibolab/result.py#L16
:DD

Reply:
ad5cfee753b2968740c42c919c8c8cad2d8fa70b